pkg/operator/ceph/file/mds.go

```diff
@@ -74,27 +77,8 @@ func newCluster(
 
 type mdsConfig struct {
 	ResourceName string
-	DaemonName   string
-}
-
-// return true if an attempt was made to prepare the filesystem associated for upgrade
-func (c *cluster) deleteLegacyMdsDeployment() bool {
-	legacyName := fmt.Sprintf("%s-%s", AppName, c.fs.Name) // rook-ceph-mds-<fsname>
-	logger.Debugf("getting legacy mds deployment %s for filesystem %s if it exists", legacyName, c.fs.Name)
-	_, err := c.context.Clientset.Apps().Deployments(c.fs.Namespace).Get(legacyName, metav1.GetOptions{})
-	if err != nil && errors.IsNotFound(err) {
-		logger.Infof("legacy mds deployment %s not found, no update needed", legacyName)
-		return false
-		// if any other error, failed to get legacy dep., but it exists, so still try deleting it.
-	}
-	logger.Infof("deleting legacy mds deployment %s", legacyName)
-	if err := mdsdaemon.PrepareForDaemonUpgrade(c.context, c.fs.Namespace, c.fs.Name, fsWaitForActiveTimeout); err != nil {
-		logger.Errorf("failed to prepare filesystem %s for mds upgrade while deleting legacy deployment %s, continuing anyway: %+v", c.fs.Name, legacyName, err)
-	}
-	if err := k8sutil.DeleteDeployment(c.context.Clientset, c.fs.Namespace, legacyName); err != nil {
-		logger.Errorf("failed to delete legacy deployment %s, USER should delete the legacy mds deployment manually if it still exists: %+v", legacyName, err)
-	}
-	return true
+	DaemonName   string              // TODO: change to 'DaemonID'
```

Comment:
I know the struct was already there but perhaps you can move it at the top since you're editing it?